### PR TITLE
Create parameters description in the documentation

### PR DESCRIPTION
**Pull request by nico-canta**
* Switch to Python 3.11
* Compile `petsc4py` manually with the correct Cython
* Mock `petsc4py` on ReadTheDocs to accelerate documentation building (takes a few minutes _less_)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,5 @@
-![AstroVascPy Logo](docs/source/logo/BBP-AstroVascPy-Github.jpg)
+<img src="https://github.com/BlueBrain/AstroVascPy/blob/main/docs/source/logo/BBP-AstroVascPy-Github.jpg?raw=true" alt="AstroVascPy Logo"/>
```
Comment (matz-e) [inline README.md:1]: Why this change? This will unconditionally refer to the image in the default branch and makes gauging changes harder.

Comment (nico-canta) [reply README.md:1]: https://github.com/BlueBrain/AstroVascPy/blob/c2ac01ad5d9d794d84ea4001dccea84948589689/docs/source/index.rst?plain=1#L1

Here we include the `readme.md`. 
But the generated website [astrovascpy.rtfd.io](https://astrovascpy.rtfd.io/) does not show the logo.  
That change fixes the problem, but I don't know if there is a better way. Any suggestion?   (also, at the moment I don't have access to the building process)

Comment (nico-canta) [reply README.md:1]: Ok, maybe I have another idea.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -7,26 +7,60 @@ on:
 env:
   apt_options: -o Acquire::Retries=3
   PETSC_DIR: /usr/lib/petscdir/petsc3.15/x86_64-linux-gnu-real
```
Comment (nico-canta) [inline .github/workflows/test.yml:9]: Can I remove this variable?

Comment (matz-e) [reply .github/workflows/test.yml:9]: It is still referenced by the installation of `petsc4py`, IIRC. So I think it should stay in.

File: astrovascpy/typing.py [hunk 3]
```diff
@@ -23,9 +23,49 @@ class VasculatureAxis(enum.IntEnum):
...
+    |  max_r_capill: max radius change factor for capillaries
+    |  t_2_max_capill: time (in seconds) to reach r_max from 0 for capillaries
+    |  max_r_artery: max radius change factor for arteries
+    |  t_2_max_artery: time (in seconds) to reach r_max from 0 for arteries
+
+    PETSc Linear solver parameters (optional):
+
+    |  solver: iterative linear solver used by PETSc
+    |  max_it: maximum number of solver iterations
+    |  r_tol: relative tolerance
```
Comment (matz-e) [inline astrovascpy/typing.py:52]: This renders like this:
image

But ideally, we'd want them to look like this:
image

Maybe something like this would be better?
```suggestion
    Required parameters used in the model are as follows:

    Args:
        vasc_axis: vasculature axis corresponding to x, y, or z. Should be set to 0, 1, or 2.
        depth_ratio: depth along the vasc_axis. This is the portion of the vasculature where there are inputs.
        max_nb_inputs: maximum number of entry nodes where we inject the flow into the vasculature. Should be >= 1.
        blood_viscosity: plasma viscosity in g.µm^-1.s^-1
        base_pressure: reference pressure in g * um^{-1} * s^{-2}. At resting state equal to the external pressure

    Optional stochastic simulation parameters can also be provided.

    Args:
        threshold_r: radius (µm) threshold. A radius smaller than the threshold is considered a capillary. A radius bigger than the threshold is considered an artery.

        c_cap: constant used in the ROU parameter calibration for capillaries
        c_art: constant used in the ROU parameter calibration for arteries

        max_r_capill: max radius change factor for capillaries
        t_2_max_capill: time (in seconds) to reach r_max from 0 for capillaries
        max_r_artery: max radius change factor for arteries
        t_2_max_artery: time (in seconds) to reach r_max from 0 for arteries

    Optional PETSc Linear solver parameters.

    Args:
        solver: iterative linear solver used by PETSc
        max_it: maximum number of solver iterations
        r_tol: relative tolerance
```
I'm not sure if it works with multiple `Args` things, but it would be worth a try IMO.

File: .github/workflows/test.yml (hunk 2 above)
Comment (nico-canta) [reply .github/workflows/test.yml:9]: ok!

File: astrovascpy/typing.py (hunk 3 above)
Comment (nico-canta) [reply astrovascpy/typing.py:52]: thanks! Now I try!

**Review by matz-e — APPROVED** [review]
